### PR TITLE
cava: add SDL2 library, Pipewire and OpenGL during build

### PR DESCRIPTION
**Pull request by Arjun31415**
## Description of changes

Cava now supports sdl2 graphical context, as shown in config file here

https://github.com/karlstav/cava/blob/81211e9e16302312ee4322a99840c110bce0e13d/example_files/config#L95-L96

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Arjun31415** [comment]
pinging maintainers @mirrexagon

**Review by NotAShelf — COMMENTED** [review]
I would say this should be called "withSDL2" instead of enable, since there is no enabling being done.

**Comment by NotAShelf** [comment]
you'll also want to squash both commits into one

**Comment by Arjun31415** [comment]
Squashed

**Review by NotAShelf — APPROVED** [review]

File: pkgs/applications/audio/cava/default.nix [hunk 1]
```diff
@@ -27,7 +29,7 @@ stdenv.mkDerivation rec {
     libpulseaudio
     ncurses
     iniparser
-  ];
+  ]++ lib.optionals withSDL2 [SDL2];
```
Comment (fufexan) [inline pkgs/applications/audio/cava/default.nix:32]:
```suggestion
  ]++ lib.optional withSDL2 SDL2;
```
Doesn't matter much to me, maybe a committer has a stronger opinion on this.

**Review by LudovicoPiero — APPROVED** [review]

**Review by fufexan — APPROVED** [review]

**Comment by Arjun31415** [comment]
Can this be merged ?

**Review by fufexan — APPROVED** [review]

**Comment by majiru** [comment]
Was there any particular reason that the pipewire support is gated behind a withPipewire here? It would be nice to have it built with support by default (or find a way to expose it better) to avoid issues like https://github.com/NixOS/nixpkgs/issues/237498 and https://github.com/NixOS/nixpkgs/issues/249876. Or is there place to document this flag to make it more obvious?

**Comment by Arjun31415** [comment]
Well I had no particular reason, and a PR to build with pipewire also is easy. Just change the default to true instead of false.

**Comment by majiru** [comment]
> Well I had no particular reason, and a PR to build with pipewire also is easy. Just change the default to true instead of false.

Thank you. I have created a PR for this here https://github.com/NixOS/nixpkgs/pull/254805

**Pull request merged**